### ISSUE TITLE
cd - does not return to previous working directory

### ISSUE THREAD

**Issue by kputland** [hyperupcall/autoenv#156]
penglund-lt:~ karl$ cd Downloads/
penglund-lt:Downloads karl$ cd -
/Users/karl/Downloads
penglund-lt:Downloads karl$ pwd
/Users/karl/Downloads
penglund-lt:Downloads karl$

**Comment by kputland** [comment]
Without autoenv in .profile I get the correct behavior

penglund-lt:~ karl$ cd Downloads/
penglund-lt:Downloads karl$ cd -
/Users/karl
penglund-lt:~ karl$ pwd
/Users/karl
penglund-lt:~ karl$

**Comment by ernstki** [comment]
I'm not able to reproduce this on OS X _or_ a Debian 8.something box that I have access to, so maybe something else funny is going on.

Out of curiosity: do you get the same behavior with a "fresh" user account? (Just create a dummy user, clone autoenv somewhere in his home directory, then source `/path/to/activate.sh` from `.bashrc`.)

**Comment by dasJ** [comment]
I can not confirm this either on Linux with a zsh

**Comment by imbible** [comment]
This happens to me on macOS El Capitan.

**Comment by eakl** [comment]
Happens to me as well. Typing '$ env' shows that $OLDPWD and $PWD are the same when autoenv is sourced.

**Comment by mohamed-aziz** [comment]
Happens to me also on archlinux with zsh.

**Comment by gergnz** [comment]
I'm affected by this. OS X El Capitan with brew bash4.

**Comment by gergnz** [comment]
This seems to be a problem in 0.2.1.
This line causes the problem: https://github.com/kennethreitz/autoenv/blob/v0.2.1/activate.sh#L14

Workaround for this version:
```diff
--- x.sh	2017-03-02 09:50:59.000000000 +1100
+++ /var/tmp/activate.sh	2017-03-02 09:48:33.000000000 +1100
@@ -2,7 +2,8 @@
 AUTOENV_ENV_FILENAME="${AUTOENV_ENV_FILENAME:-.env}"

 autoenv_init() {
-	local _mountpoint _files _orderedfiles _sedregexp _pwd
+	local _mountpoint _files _orderedfiles _sedregexp _pwd _oldpwd
+  _oldpwd=${OLDPWD}
 	if [ "${OSTYPE#darwin*}" != "${OSTYPE}" ]; then
 		_sedregexp='-E'
 	else
@@ -63,6 +64,7 @@
 	if [ -z "${zsh_shwordsplit}" ]; then
 		unsetopt shwordsplit >/dev/null 2>&1
 	fi
+  OLDPWD=${_oldpwd}
 }

 autoenv_hashline() {
```

Or pull the master (mac OS brew...):
```bash
curl -o /usr/local/opt/autoenv/activate.sh https://raw.githubusercontent.com/kennethreitz/autoenv/master/activate.sh
```

Seems to be related to #151 which was fixed by #157

**Comment by grayside** [comment]
Seeing this on OSX El Capitan with Zsh 5.3.

**Comment by ddaza** [comment]
I see this behavior in OSX in `GNU bash, version 4.4.12(1)-release (x86_64-apple-darwin16.3.0)`

**Comment by williamsmj** [comment]
This was fixed in #149.

The most recent tagged release predates #149, so current packaged up versions (including brew) have this bug. Not sure if there are plans for a tagged release any time soon, but in the mean time copy `https://raw.githubusercontent.com/kennethreitz/autoenv/master/activate.sh` over your installed version.

**Comment by Gnouc** [comment]
Can we close this issue now?

**Comment by dapicester** [comment]
Waiting for a new release, for those using `brew` just install the `HEAD` version to get the fix:
```shell
$ brew install autoenv --HEAD
```

**Comment by hyperupcall** [comment]
This fix has been released in [v0.3.0](https://github.com/hyperupcall/autoenv/releases/tag/v0.3.0)! Closing
